### PR TITLE
Fix `AutomaticSpeechRecognitionPipelineTests.run_pipeline_test`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix `AutomaticSpeechRecognitionPipelineTests.run_pipeline_test` which was changed in #19570 and #20104.

See the comments in this PR changes.

I detected this when working on improving pipeline tests using tiny models. Previously, `Speech2TextConfig` is not used in ASR pipeline tests, but now it does, and  gives errors without this PR.

### CODE REVIEW

File: tests/pipelines/test_pipelines_automatic_speech_recognition.py [hunk 1]
```diff
@@ -118,15 +118,10 @@ def run_pipeline_test(self, speech_recognizer, examples):
...
-            # Non CTC models cannot use chunk_length
-            with self.assertRaises(ValueError) as v:
-                outputs = speech_recognizer(audio, chunk_length_s=10)
-            self.assertEqual(v.exception, "")
-
             # Non CTC models cannot use return_timestamps
             with self.assertRaises(ValueError) as v:
                 outputs = speech_recognizer(audio, return_timestamps="char")
-            self.assertEqual(v.exception, "")
+            self.assertEqual(str(v.exception), "We cannot return_timestamps yet on non-ctc models !")
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_automatic_speech_recognition.py:124]: I don't understand why we expected empty string here, while we have

https://github.com/huggingface/transformers/blob/67d32f4649715db5cbac66deb759f9969dc1ff70/src/transformers/pipelines/automatic_speech_recognition.py#L405

@Narsil

File: tests/pipelines/test_pipelines_automatic_speech_recognition.py [hunk 2]
```diff
@@ -118,15 +118,10 @@ def run_pipeline_test(self, speech_recognizer, examples):
                 },
             )
         else:
-            # Non CTC models cannot use chunk_length
-            with self.assertRaises(ValueError) as v:
-                outputs = speech_recognizer(audio, chunk_length_s=10)
-            self.assertEqual(v.exception, "")
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_automatic_speech_recognition.py:124]: PR #20104 have the following change (see [here](https://github.com/huggingface/transformers/commit/25c451e5a044969eb91e1e481574a2bfca5130ca#diff-ccc8e98fcaa81fdf6317a652438a309bcede0bbe336774288c2fcf91d9f11082)), and we don't expect to have exception here anymore

from
```python
        if chunk_length_s:
            if self.type not in {"ctc", "ctc_with_lm"}:
                raise ValueError(
                    "`chunk_length_s` is only valid for CTC models, use other chunking options for other models"
                )
```
to
```python
            if self.type == "seq2seq" and not ignore_warning:
                logger.warning(
                    "Using `chunk_length_s` is very experimental with seq2seq models. The results will not necessarily"
                    " be entirely accurate and will have caveats. More information:"
                    " https://github.com/huggingface/transformers/pull/20104. Ignore this warning with pipeline(...,"
                    " ignore_warning=True)"
                )
```

File: tests/pipelines/test_pipelines_automatic_speech_recognition.py (hunk 1 above)
Comment (sgugger) [inline tests/pipelines/test_pipelines_automatic_speech_recognition.py:124]: Why not use `self.assertRaisesRegex` here to test the exception and the message together?

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

File: tests/pipelines/test_pipelines_automatic_speech_recognition.py (hunk 1 above)
Comment (ydshieh) [reply tests/pipelines/test_pipelines_automatic_speech_recognition.py:124]: Can do that - I just followed what has been there.

Comment (Narsil) [reply tests/pipelines/test_pipelines_automatic_speech_recognition.py:124]: Nice catch. How was this not caught earlier ?

And for the empty string, it seems likely I forgot to update it.
Usually in my workflow, I create the failing test, do the scaffolding on how the test should look like, and updte the values when the test fails.

It seems I forgot to run/update this one.

**Review by Narsil — APPROVED** [review]

**Comment by Narsil** [comment]
Thank you !

File: tests/pipelines/test_pipelines_automatic_speech_recognition.py (hunk 1 above)
Comment (ydshieh) [reply tests/pipelines/test_pipelines_automatic_speech_recognition.py:124]: > How was this not caught earlier ?

I only found this after trying to use new tiny models. Previously, `Speech2TextConfig` would not generate a pipeline test (yeah, you know, there were a lot of reasons where a tiny model wouldn't be created). In the new tiny model creation script, I made more efforts to create those tiny models as many as possible.

> And for the empty string, it seems likely I forgot to update it.
No problem :-) I also guess so

Comment (ydshieh) [reply tests/pipelines/test_pipelines_automatic_speech_recognition.py]: Done!

**Pull request merged**